alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs

```diff
@@ -312,10 +317,19 @@ utxoTransition = do
   let Shelley.UTxOState utxo _deposits _fees _ppup = u
 
   let txb = txbody tx
+  let vi@(ValidityInterval _ i_f) = getField @"vldt" txb
 
-  inInterval slot (getField @"vldt" txb)
+  inInterval slot vi
     ?! OutsideValidityIntervalUTxO (getField @"vldt" txb) slot
 
+  sysSt <- liftSTS $ asks systemStart
+  ei <- liftSTS $ asks epochInfoWithErr
+  case i_f of
+    SNothing -> pure ()
+    SJust ifj -> case (epochInfoSlotToUTCTime ei sysSt ifj) of
+      Left _ -> False ?!# OutsideForecast ifj -- error translating slot
```

Comment:
you can use `failBecause` from `Control.State.Transition`. (it's the same thing, just reads nicer.)

Reply:
thanks!